src/Monitoring/Sdk/GrafanaSerialization.cs

```diff
@@ -108,6 +118,118 @@ public static JObject SanitizeNotificationChannel(JObject notificationChannel)
             notificationChannel.Remove("updated");
             return notificationChannel;
         }
+
+        public static JObject ParameterizeDashboard(JObject dashboard, ICollection<Parameter> parameters, IEnumerable<string> environments, string activeEnvironment)
+        {
+            JObject parameterizedDashboard = new JObject(dashboard);
+
+            // These paths may be parameterized
+            string[] paths = new[]
+            {
+                "$.dashboard.panels[*].targets[*].azureMonitor.resourceGroup",
+                "$.dashboard.panels[*].targets[*].azureMonitor.resourceName",
+                "$.dashboard.panels[*].targets[*].azureLogAnalytics.resource",
+                "$.dashboard.panels[*].targets[*].azureLogAnalytics.workspace"
+            };
+
+            foreach (string path in paths)
+            {
+                IEnumerable<JToken> tokens = parameterizedDashboard.SelectTokens(path);
+
+                foreach (JToken token in tokens)
+                {
+                    string value = token.Value<string>();
+
+                    // Value Can technically be other JSON types (but the paths specified above
+                    // will always be strings). Because the object must be a "string" to store the
+                    // parameter info, supporting anything other than a string will require also
+                    // keeping type info. So only support strings for now.
+
+                    if (value == null)
+                    {
+                        throw new Exception("Unexpected token type. Value must be a string.");
+                    }
+
+                    Parameter p = parameters
+                        .Where(p => p.Values.ContainsKey(activeEnvironment) && p.Values[activeEnvironment] == value)
+                        .FirstOrDefault();
+
+                    if (p == null)
+                    {
+                        string name = Guid.NewGuid().ToString();
+
+                        p = new Parameter()
+                        {
+                            Name = name,
+                            Values = environments.Select(env =>
+                            {
+                                if (env == activeEnvironment)
+                                {
+                                    return (env, value);
+                                }
+                                else
+                                {
+                                    return (env, "PLACEHOLDER");
+                                }
+                            }).ToDictionary(e => e.env, e => e.Item2)
```

Comment:
Why do the select then the ToDictionary, why not just a to dictionary and pick the value with the if statement?  (That would avoid the weird "Item2" floating around)